cmd/security-secrets-setup/main_test.go

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/setup"
+	"github.com/edgexfoundry/edgex-go/internal/security/setup/option"
+	"github.com/stretchr/testify/assert"
+)
+
+type testExitCode struct {
+	testStatusCode int
+}
+
+type testPkiInitOptionDispatcher struct {
+	testOptsExecutor option.OptionsExecutor
+}
+
+var hasDispatchError bool
+
+func TestMainWithNoOption(t *testing.T) {
+	tearDown := setupTest(t)
+	origArgs := os.Args
+	defer tearDown(t, origArgs)
+	assert := assert.New(t)
+
+	runWithNoOption()
+	assert.Equal(1, (exitInstance.(*testExitCode)).getStatusCode())
+}
+
+func TestMainWithConfigFileOption(t *testing.T) {
+	tearDown := setupTest(t)
+	origArgs := os.Args
+	defer tearDown(t, origArgs)
+	assert := assert.New(t)
+
+	runWithConfigFileOption()
+	assert.Equal(0, (exitInstance.(*testExitCode)).getStatusCode())
+
+	// verify ./config/pki/EdgeXFoundryCA directory exists
+	exists, err := doesDirectoryExist("./config/pki/EdgeXFoundryCA")
+	if !exists {
+		assert.NotNil(err)
+		assert.FailNow("cannot find the directory for TLS assets", err)
+	}
+}
+
+func runWithNoOption() {
+	// case 1: no option given
+	os.Args = []string{"cmd"}
+	printCommandLineStrings(os.Args)
+	main()
+}
+
+func runWithConfigFileOption() {
+	os.Args = []string{"cmd", "-config", "./res/pkisetup-vault.json"}
+	printCommandLineStrings(os.Args)
+	main()
+}
+
+func printCommandLineStrings(strs []string) {
```

Comment:
it seems like this entire function could just be `fmt.Println("command line strings:",strings.Join(strs," "))`

Reply:
ok